src/system/SystemConfig.h

```diff
@@ -806,3 +806,57 @@ struct LwIPEvent;
 #ifndef CHIP_SYSTEM_CONFIG_MAX_LARGE_BUFFER_SIZE_BYTES
 #define CHIP_SYSTEM_CONFIG_MAX_LARGE_BUFFER_SIZE_BYTES (64000)
 #endif
+
+/**
+ * @file
+ * Configuration settings for Terms and Conditions (TC) acknowledgements during device commissioning.
+ */
+
+/**
+ * @def CHIP_CONFIG_TC_REQUIRED
+ *
+ * @brief Indicates whether terms and conditions are required during commissioning.
+ *
+ * This macro defines whether the device commissioning process requires the user to acknowledge terms and conditions.
+ * - 1: Terms and conditions are required.
+ * - 0: Terms and conditions are not required.
+ *
+ * If this is set to 1, both CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS and
+ * CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS_VERSION must be defined.
+ */
+#ifndef CHIP_CONFIG_TC_REQUIRED
+#define CHIP_CONFIG_TC_REQUIRED (0)
+#endif
```

Comment:
This is not where this belongs. This belongs in `src/lib/core/CHIPConfig.h` like all other product configs.